File: ios/react-native-bluetooth-cross-platform/MessageEncoder.swift
import Foundation
import Underdark

public protocol MessageEncoder {
  func sendMessage(message: String, userId:String)
  func sendMessage(message: String, link:UDLink)
  func informConnected(user: User)
  func informDisonnected(user: User)
  func informAcceptedInvite(user: User)
  func inviteUser(user: User)
  func broadcastType()
}

File: android/src/main/java/com/rctunderdark/MessageEncoder.java
package com.rctunderdark;

import io.underdark.transport.Link;

public interface MessageEncoder {
    public void informConnected(User user);
    public void informAccepted(User user);
    public void informDisconnected(User user);
    public void informAcceptedInvite(User user);
    public void inviteUser(User user);
    public void broadcastType();
    public void sendMessage(String message, Link link);
    public void sendMessage(String message, String id);
}
